docs/proposals/Tag-Digest-CoExist.md

```diff
@@ -0,0 +1,88 @@
+
+Other user scenarios that encompass tag mutations NOT related to the current issue which users might want:
+1. In case of the pod spec already having both tag and digest, I want to be able to mutate the tag regardless and update the digest accordingly instead of leaving the digest in place.
+
+## Solution Overview
+
+The current solution has been chosen on the basis that Ratify is only meant to mutate from tags to digest as tags are mutable and digests are the ultimate source of truth. This solution is also prepared with the community recommendation of using digests instead of tags in mind with digests being the ultimate source of truth for artifact verification. 
+
+However with the digest-only approach having altercations with broader software engineers NOT focused towards security, embedding digests alongside pre-existing tags in the K8s object spec during mutation as a debug-friendly and engineer friendly way forward seems feasible. As the end container orchestration framework such as `containerd` and ultimately `runc` still continue to rely on only the mutated digest to create containers, engineers on the other hand can rely on the pre-existing and untouched tag in the deployed object (Deployment, Pod, StatefulSet etc)’s image spec to know their source of truth for debugging purposes. 
+
+As discussed in the corresponding [Github issue](https://github.com/ratify-project/ratify/issues/1657),  having both tag & digest (`<image>:<tag>@<digest>`) is NOT a recommended option but retains status for backward compatibility, new options with default configuration adhering to this shall be discussed in the design section.
+
+## Solution Design and Configurations / Proposed Changes
+
+This section discusses the various additions to the current Ratify Mutator and the corresponding Helm configuration that can help resolve this problem. We also discuss features that the mutator could incorporate to facilitate other additional problems but is a topic of further discussion and debate not within the realm and scope of the problem at hand.
+
+The solution proposes tweaks to the following sections to fix the problem at hand:
+1. The helm chart shall be changed to add the corresponding new configs facilitating these new features for mutation.
+1. The new configs shall be respectively passed to `/app/ratify serve` called through the `deployment.yaml` file to then handle the additional configs.
+1. These configs then trickle down to the corresponding mutation code block to handle the mutations according to those config.
+
+### Configurations
+
+A new config block shall be added in the helm chart’s `values.yaml` which will be used respectively. 
+
+This feature will be available through the new `provider.mutation` config block which will make the `provider.enableMutation` option obsolete. A new boolean sub-config of `provider.mutation.enable` will be added to facilitate this existing feature.
```

Comment:
We will need to keep the `enableMutation` in the helm chart till the next major release (v2.0.0) as this is a breaking change. In the mean time, we can mark the value as deprecated and update the documentation accordingly to use the new `provider.mutation.enabled` flag instead. However, functionally both values must work.

Reply:
Gotcha. I've updated the proposal to sync with this.